src/utils/fastmath.h

```diff
@@ -82,9 +82,21 @@ inline float FastLogit(const float a) {
   return 0.5f * FastLog((1.0f + a) / (1.0f - a));
 }
 
-// Fast approximate a^x.
+// Fast approximate x^y. Works for x>0 and saturates on underflow.
+// The approximation used here is a combination of log2(2^N*(1+f)) ~ N+f+k
+// and 2^(N+f) ~ 2^N*(1+f-k) with N integer and f the fractional part, f>=0.
+// The constant k is used to tune the approximation accuracy.
```

Comment:
what constant k?

Reply:
I'm guessing a magical constant that you can tune to optimize accuracy for a particular x,y distribution. The maximal error will depend on what range you expect to see.

Reply:
right, but the code should probably have a constant k, not just the comment.

Reply:
This is the high level description of the algorithm, similar to the ones earlier in the file. Are you asking for the exact value of the constant or to see it in the code (where it is indeed folded)? BTW in this case there is nothing magical about it, the error of the approximation is very one sided, so the constant is just an offset added to make it symmetrical. I expect there is still some improvement possible by tuning the two k independently (they are identical now) but this has no effect on the speed so is not done yet.